### PR TITLE
feat: Add options to redact or ignore view for Replay

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Added options to ReplayOptions for users to specify which classes to redact or ignore during replay. 
Also added functions to SentrySDK and a UIView extension to choose specific views to redact or ignore.

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Review by philipphofmann — APPROVED** [review]
LGTM, with a couple of comments. Nothing blocking.

File: Sources/Sentry/Public/SentrySDK.h [hunk 1]
```diff
@@ -333,6 +334,21 @@ SENTRY_NO_INIT
  */
 + (void)close;
 
+#if SENTRY_HAS_UIKIT
+
+/**
+ * Marks this view to be redacted during replays.
+ */
++ (void)replayRedactView:(UIView *)view;
```
[the remark this answers is not in the document]
Comment (brustolin) [reply Sources/Sentry/Public/SentrySDK.h:342]: Users may use a view as container to mask/ignore all its content without subclassing it.

File: Sources/Swift/Extensions/UIViewExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#if canImport(UIKit) && !SENTRY_NO_UIKIT
+#if os(iOS) || os(tvOS)
+import Foundation
+import UIKit
+
+public extension UIView {
```
[the remark this answers is not in the document]
Comment (brustolin) [reply Sources/Swift/Extensions/UIViewExtensions.swift:6]: We're giving this extensions to users. This is not for the SDK to use.

File: Sources/Swift/Tools/UIRedactBuilder.swift [hunk 3]
```diff
@@ -68,10 +69,35 @@ class UIRedactBuilder {
...
 #else
-        ignoreClasses = []
+        ignoreClassesIdentifiers = []
 #endif
+        redactClassesIdentifiers = Set(redactClasses.map( { ObjectIdentifier($0) }))
+    }
+    
+    func hasIgnoreClass(_ ignoreClass: AnyClass) -> Bool {
+        return ignoreClassesIdentifiers.contains(ObjectIdentifier(ignoreClass))
+    }
```
[the remark this answers is not in the document]
Comment (brustolin) [reply Sources/Swift/Tools/UIRedactBuilder.swift:81]: IMO the use of insert for Sets is awful. Insert in most collections/languages implies that you can choose the position. I rather keep it as `add`.

File: Sources/Sentry/Public/SentrySDK.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/Public/SentrySDK.h:344]: Ah that makes sense thanks for explaining.

File: Sources/Swift/Extensions/UIViewExtensions.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Swift/Extensions/UIViewExtensions.swift:6]: Then please again mark this as experimental in the code comment.

**Review by philipphofmann — APPROVED** [review]
Again, LGTM.

File: Sources/Swift/Extensions/UIViewExtensions.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+     * Marks this view to be redacted during replays.
+     */
+    func sentryReplayRedact() {
+        SentryRedactViewHelper.redactView(self)
+    }
+    
+    /**
+     * Marks this view to be ignored during redact step
+     * of session replay. All its content will be visible in the replay.
+     */
```
Comment (brustolin) [inline Sources/Swift/Extensions/UIViewExtensions.swift:18]:
```suggestion
    /**
     * Marks this view to be ignored during redact step
     * of session replay. All its content will be visible in the replay.
     * - warning:  This is an experimental feature and may still have bugs.
     */
```

**Pull request merged**